### PR TITLE
More virtual memory checks and bounds

### PR DESCRIPTION
**Pull request by qwattash**

### CODE REVIEW

File: sys/vm/vm_page.h [hunk 1]
```diff
@@ -1012,6 +1013,34 @@ vm_page_domain(vm_page_t m)
 #endif
 }
 
+static inline vm_page_t
+vm_page_array_slice(vm_page_t m, int npages)
+{
+
```
Comment (brooksdavis) [inline sys/vm/vm_page.h:1019]:
```suggestion
```
Blank line no longer required so might as well skip it.

**Review by brooksdavis — APPROVED** [review]
Looks good overall.

File: sys/kern/subr_vmem.c [hunk 2]
```diff
@@ -1039,14 +1039,15 @@ vmem_fit(const bt_t *bt, vmem_size_t size, vmem_size_t align,
 		MPASS(align < nocross);
 		start = VMEM_ALIGNUP(start - phase, nocross) + phase;
 	}
-	if (start <= end && (ptraddr_t)end - (ptraddr_t)start >= size - 1) {
-		MPASS(((ptraddr_t)start & (align - 1)) == phase);
+
```
Comment (jrtc27) [inline sys/kern/subr_vmem.c:1042]: There's no blank line upstream

File: sys/vm/vm_glue.c [hunk 3]
```diff
@@ -525,7 +525,8 @@ kstack_cache_init(void *null)
 	    atop(VM_MAX_KERNEL_ADDRESS - VM_MIN_KERNEL_ADDRESS));
 	vm_object_set_flag(kstack_object, OBJ_HASCAP);
 	kstack_cache = uma_zcache_create("kstack_cache",
-	    kstack_pages * PAGE_SIZE, NULL, NULL, NULL, NULL,
+	    (kstack_pages + KSTACK_GUARD_PAGES) * PAGE_SIZE,
```
Comment (jrtc27) [inline sys/vm/vm_glue.c:528]: Upstreamable?

File: sys/kern/kern_sysctl.c [hunk 4]
```diff
@@ -84,7 +84,6 @@ __FBSDID("$FreeBSD$");
 
 static MALLOC_DEFINE(M_SYSCTL, "sysctl", "sysctl internal magic");
 static MALLOC_DEFINE(M_SYSCTLOID, "sysctloid", "sysctl dynamic oids");
-static MALLOC_DEFINE(M_SYSCTLTMP, "sysctltmp", "sysctl temp output buffer");
```
[the remark this answers is not in the document]
Comment (qwattash) [reply sys/kern/kern_sysctl.c:87]: I have not been able to understand how this is not an issue upstream. I don't think it is a qemu quirk on how it passes kernel parameters, but is very easy to trigger with `-append`. I should see if I can reproduce it with a freebsd build.

File: sys/vm/vm_page.c [hunk 5]
```diff
@@ -222,7 +222,7 @@ vm_page_init_cache_zones(void *dummy __unused)
 			pgcache->domain = domain;
 			pgcache->pool = pool;
 			pgcache->zone = uma_zcache_create("vm pgcache",
-			    PAGE_SIZE, NULL, NULL, NULL, NULL,
+			    sizeof(struct vm_page), NULL, NULL, NULL, NULL,
```
Comment (jrtc27) [inline sys/vm/vm_page.c:225]: Upstreamable?

File: sys/vm/vm_reserv.c [hunk 6]
```diff
@@ -927,6 +933,7 @@ vm_reserv_alloc_page(vm_object_t object, vm_pindex_t pindex, int domain,
 		}
 	} else
 		return (NULL);
+
```
Comment (jrtc27) [inline sys/vm/vm_reserv.c:936]:
```suggestion
```

File: sys/vm/vm_glue.c (hunk 3 above)
Comment (qwattash) [reply sys/vm/vm_glue.c]: Moved to the upstreamable PR  #1008 to keep track.

File: sys/vm/vm_page.c (hunk 5 above)
Comment (qwattash) [reply sys/vm/vm_page.c]: Moved to the upstreamable PR  #1008 to keep track.

File: sys/kern/kern_sysctl.c (hunk 4 above)
[the remark this answers is not in the document]
Comment (qwattash) [reply sys/kern/kern_sysctl.c:87]: Ok so this issue was fixed upstream by https://github.com/freebsd/freebsd-src/commit/4342ba184c15 I will cherry-pick that instead.

**Pull request merged**